app/services/puppet_fact_parser.rb

```diff
@@ -145,6 +145,14 @@ def disks_total
     facts['disks']&.values&.sum { |disk| disk&.fetch('size_bytes', 0).to_i }
   end
 
+  def self.smart_proxy_features
+    'Puppet'.freeze
```

Comment:
I don't think this is correct and that the Puppet feature on the Smart Proxy doesn't have anything to do with fact parsing. It is only an abstraction over Puppetserver's API (and translates the authentication) but all communication from Puppetserver back to Foreman happens directly via the API.

On the other hand, is this used to check if the Smart Proxy is authorized to send facts? If so, I'd try to include that in the method name to make it clear.

Reply:
Well, my idea behind moving this into parser is the fact that when I have a parser, I can accept that kind of fact. Also it was originally at importer and this is only parser specific place.